app/protocol/flows/blockrelay/ibd.go

```diff
@@ -366,67 +378,78 @@ func (flow *handleRelayInvsFlow) fetchMissingUTXOSet(ibdRootHash *externalapi.Do
 	return true, nil
 }
 
-func (flow *handleRelayInvsFlow) receiveIBDRootUTXOSetAndBlock() ([]byte, *externalapi.DomainBlock, bool, error) {
-	onEnd := logger.LogAndMeasureExecutionTime(log, "receiveIBDRootUTXOSetAndBlock")
+func (flow *handleRelayInvsFlow) receiveIBDRootBlock() (*externalapi.DomainBlock, error) {
+	onEnd := logger.LogAndMeasureExecutionTime(log, "receiveIBDRootBlock")
 	defer onEnd()
 
 	message, err := flow.dequeueIncomingMessageAndSkipInvs(common.DefaultTimeout)
 	if err != nil {
-		return nil, nil, false, err
+		return nil, err
 	}
 
-	var block *externalapi.DomainBlock
-	switch message := message.(type) {
-	case *appmessage.MsgIBDBlock:
-		block = appmessage.MsgBlockToDomainBlock(message.MsgBlock)
-	case *appmessage.MsgIBDRootNotFound:
-		return nil, nil, false, nil
-	default:
-		return nil, nil, false,
-			protocolerrors.Errorf(true, "received unexpected message type. "+
-				"expected: %s or %s, got: %s",
-				appmessage.CmdIBDBlock, appmessage.CmdIBDRootNotFound, message.Command(),
-			)
+	ibdBlockMessage, ok := message.(*appmessage.MsgIBDBlock)
+	if !ok {
+		return nil, protocolerrors.Errorf(true, "received unexpected message type. "+
+			"expected: %s, got: %s", appmessage.CmdIBDBlock, message.Command())
 	}
+	block := appmessage.MsgBlockToDomainBlock(ibdBlockMessage.MsgBlock)
+
 	log.Debugf("Received IBD root block %s", consensushashing.BlockHash(block))
 
-	serializedUTXOSet := []byte{}
+	return block, nil
+}
+
+func (flow *handleRelayInvsFlow) receiveAndInsertIBDRootUTXOSet() (bool, error) {
+	onEnd := logger.LogAndMeasureExecutionTime(log, "receiveAndInsertIBDRootUTXOSet")
+	defer onEnd()
+
 	receivedAllChunks := false
 	receivedChunkCount := 0
+	receivedUTXOCount := 0
```

Comment:
What's stopping the other node from similarly lying about the amount of UTXOs?
This would probably require some thinking. Let's open a ticket for it.